core/web/job_runs_controller.go

```diff
@@ -54,17 +56,42 @@ func (jrc *JobRunsController) Create(c *gin.Context) {
 		jsonAPIError(c, http.StatusNotFound, errors.New("Job not found"))
 	} else if err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
-	} else if !j.WebAuthorized() {
-		jsonAPIError(c, http.StatusForbidden, errors.New("Job not available on web API, recreate with web initiator"))
+	} else if initiator, err := getAuthenticatedInitiator(c, j); err != nil {
+		jsonAPIError(c, http.StatusForbidden, err)
 	} else if data, err := getRunData(c); err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
-	} else if jr, err := services.ExecuteJob(j, j.InitiatorsFor(models.InitiatorWeb)[0], models.RunResult{Data: data}, nil, jrc.App.GetStore()); err != nil {
+	} else if jr, err := services.ExecuteJob(j, *initiator, models.RunResult{Data: data}, nil, jrc.App.GetStore()); err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
 	} else {
 		jsonAPIResponse(c, presenters.JobRun{JobRun: *jr}, "job run")
 	}
 }
 
+// getInitiator returns the Job Spec's initiator for the given web context.
+func getAuthenticatedInitiator(c *gin.Context, js models.JobSpec) (*models.Initiator, error) {
+	if _, ok := authenticatedUser(c); ok {
+		webInitiators := js.InitiatorsFor(models.InitiatorWeb)
+		if len(webInitiators) == 0 {
+			return nil, errors.New("Job not available on web API, recreate with initiator type 'InitiatorWeb'")
+		}
+		return &webInitiators[0], nil
+	} else if ei, ok := authenticatedEI(c); ok {
+		var initiator *models.Initiator
+		for _, i := range js.InitiatorsFor(models.InitiatorExternal) {
+			if strings.ToLower(i.Name) == ei.Name {
```

Comment:
Should we also be testing for `type == InitiatorExternal` ? Might want to put this logic into a function in EA's packages.

Reply:
Is the suggestion to add a condition in the 'if' statement? it so, it would be redundant.  The InitiatorFor(..) above only returns Initiators of the InitiatorExternal type.

Reply:
What about js.ExternalInitiator(name)?

Reply:
I've added JobSpec.InitiatorExternal(..) to handle this case.